app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,71 @@
+class ReportsController < ApplicationController
+  before_action :set_report, only: %i[ show edit update destroy ]
+
+  # GET /reports or /reports.json
+  def index
+    @reports = Report.all
+  end
+
+  # GET /reports/1 or /reports/1.json
+  def show
+  end
+
+  # GET /reports/new
+  def new
+    @report = Report.new
+  end
+
+  # GET /reports/1/edit
+  def edit
+  end
+
+  # POST /reports or /reports.json
+  def create
+    hoge = report_params
+    hoge["user_id"] = current_user.id
+    @report = Report.new(hoge)
```

Comment:
```suggestion
    @report = Report.new(hoge){ _1.user = currentuser }
```

Reply:
前提としてユーザー入力値（params）とサーバー設定値は分離すべき、というのは口頭で @morimorihoge さんに話してもらったとおりとして

具体的なスニペットレベルでは、 `collection.build` を使うのでもいいです

https://railsguides.jp/association_basics.html#has-manyで追加されるメソッド-collection-build-attributes

```suggestion
    @report = current_user.reports.build()
```

Reply:
コメントありがとうございます。

`build`使う方法が個人的にやりたいことを直接的にできている気がするので、そちらで書こうと思います。
ちなみに、現在は`new`と違いはないみたいですが、この場合`build`を使うのがお作法と思っていて間違いないでしょうか？

Reply:
違いがないのはあってます🙆（`new` は `build` の[エイリアス](https://api.rubyonrails.org/v7.0/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-new)）

この場合の作法なのかどうかについては自信ないですが、
作法かどうかというより、[公式ドキュメントであるRailsガイドでbuildを使う説明になっている](https://railsguides.jp/association_basics.html#has-many%E3%81%A7%E8%BF%BD%E5%8A%A0%E3%81%95%E3%82%8C%E3%82%8B%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89-collection-build-attributes)ので、そこに従ってnewよりbuildかなあという感覚でした

Reply:
個人的には、`new` を使って書かれてもレビューで弾くほどではないです